src/Data/DefaultDataProvider.php

```diff
@@ -548,157 +557,176 @@ protected function updateModelInDatabase($model)
     {
         $data = $this->convertModelToDataPropertyArray($model);
 
-        $this->objDatabase
-            ->prepare(sprintf('UPDATE %s %%s WHERE id=?', $this->strSource))
-            ->set($data)
-            ->execute($model->getId());
+        $this->connection->update($this->source, $data, [['id' => $model->getId()]]);
```

Comment:
Are you sure that the double array syntax is correct here

Reply:
You're right about that. I'm changing that.